cmake/EthCompilerSettings.cmake

```diff
@@ -197,6 +197,8 @@ if (SANITIZE)
 	# -fsanitize-address-use-after-scope throws an error when a variable is used beyond its scope
 	if (san STREQUAL "address")
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-omit-frame-pointer -fsanitize=address -fsanitize-address-use-after-scope")
+	elseif (san STREQUAL "undefined")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr -fno-sanitize-recover=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unreachable,vla-bound,vptr")
```

Comment:
You specify `unsigned-integer-overflow` with `-fno-sanitize-recover` but not with `-fsanitize`. Is it a mistake and you just forgot to add it for `-fsanitize` or is it on purpose? If on purpose, this definitely needs a comment.

Reply:
> You specify unsigned-integer-overflow for -fno-sanitize-recover but not for -fsanitize

I think it is the other way round, right?

>  If on purpose, this definitely needs a comment.

Yup, agree. Will do.

Reply:
> I think it is the other way round, right?

Ah, true. Sorry. I must have looked at modified version that I've been in the process of editing locally to create the suggestion below :)

Makes sense then. But is that really the only one we want recovery for? Are the others too spammy if you let them continue?